hooks/regenerage-spm-package-if-needed.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Was there a new file added or deleted?
+if [ ! -z "$(git diff --staged | grep  -e 'deleted\|new file mode')" ];
+then
+    # Regenerate list of files in Package.swift
+    ./GenerateSPMFileLists.swift
+
+    # Add changes to the current commit 
```

Comment:
This solution is a bit naive. It can happen that also the changes that weren't staged will be added. We can improve later if it's a problem.